### ISSUE TITLE
Vaccination Certificate Validity Check: Confusing wording / different results between CWA and CovPass

### ISSUE THREAD

**Issue by Jo-Achim** [corona-warn-app/cwa-app-android#4053]
or
_It's about the "validity of certificates" vs. "unverifiability" in the absence of entry rules._

## Avoid duplicates
* [X] Bug is not mentioned in the [FAQ](https://www.coronawarn.app/en/faq/)
* [ ] Bug is specific for Android only ????, for general issues / questions that apply to iOS and Android please raise them in the [documentation repository](https://github.com/corona-warn-app/cwa-documentation)
* [X] Bug is not already reported in another issue

## Technical details

- CWA-App version: 2.9.0
- CovPass-App version: 1.95.12

## Describe the bug

Hi there,

the different statements between the **CWA** and **CovPass** seem to contradict each other or are somewhat poorly worded.

Example for Italy, entry on 12.09.2021 12:00 p.m., checked on 11.09.2021 ~ 2 p.m.:
If you carry out a validity check for a certificate with **CovPass**, you get, for example:
"**This certificate is valid on the travel date.** For Italy, 12.09.2021 12:00
_There are currently no entry rules for this type of certificate for the selected country and time._"

If you check the same certificate in **CWA** you get:
"**Certificate cannot be checked.**
Entry to Italy on September 12th, 2021 12:00, checked on September 11th, 3:53 pm
_Your certificate could not be checked ... There are currently no entry rules ... for this type of certificate._"

During CovPass "This certificate is valid on the travel date." reports because there are no entry rules, the CWA reports that the certificate cannot be verified for the very same reason and certificate.
This is confusing for me!

Shouldn't CovPass make the more correct statement with the 'validity on the day of travel' if there are no entry rules? _Because if there are no entry rules and the checked certificate is valid (EU-wide), then nothing should stand in the way of entry!?_
If this is correct, the validity should also be clearly stated in the CWA information. 
Note: This is not about the further information under "Notes" and its sub-items in the CWA.

## Steps to reproduce the issue

Verification of the same vaccination certificate under CWA and CovPass. 

Best regards, Joachim.

----
Internal Tracking-ID: [EXPOSUREAPP-9760](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-9760)

**Comment by Ein-Tim** [comment]
Related issues: 
- https://github.com/Digitaler-Impfnachweis/covpass-ios/issues/42
- https://github.com/Digitaler-Impfnachweis/covpass-android/issues/67

There is also valuable information in this PR: https://github.com/corona-warn-app/cwa-website/pull/1683

**Comment by MikeMcC399** [comment]
@Jo-Achim 
I don't see the text in CWA 2.9.0 which you quoted. Perhaps you translated from German into English?

> "Certificate cannot be checked.
Entry to Italy on September 12th, 2021 12:00, checked on September 11th, 3:53 pm
Your certificate could not be checked ... There are currently no entry rules ... for this type of certificate."

![image](https://user-images.githubusercontent.com/66998419/132953806-2903cf34-4249-48d4-8b7e-deb39d93f9be.png)

As @Ein-Tim already stated, there is an open issue in CovPass https://github.com/Digitaler-Impfnachweis/covpass-android/issues/67 which I submitted there and they have said they will fix it (see https://github.com/Digitaler-Impfnachweis/covpass-android/issues/67#issuecomment-912267298).

Do you see an open issue with CWA? If not, then you should close this issue and follow up with CovPass https://github.com/Digitaler-Impfnachweis/covpass-android/issues/67.

**Comment by Jo-Achim** [comment]
Well, maybe. But right now the problem exists.

**And sorry, I don't presume to decide which app makes which wrong statements. I am definitely not the expert here!

But it seems logical to me that if there are currently no entry restrictions, a valid EU-wide certificate must also suffice for entry.**

But ok, I'll think about it again in the next few days.

However, I see big problems. We are increasingly moving away from 'keep is simple' to 'kindly research on different platforms / in different apps' and make sure that you find / use the right one.
**That can not be!**

It leads to uncertainty, frustration, daily research (local / national) with regard to the applicable framework conditions and the question of what or whom I can trust or what I can rely on. And so on.
If we continue like this - and please from the point of view of the **_average consumer_** - then we can close soon. Incidentally, with the consistency in terms of the usefulness / value / reliability of the respective app.

(And yes, it is becoming increasingly impossible to verify whether a topic or parts of a topic have been dealt with somewhere in the vastness of the CWA / CovPass threads or not. At least for people who do not follow this topic 24/7 either.)

**Comment by MikeMcC399** [comment]
@Jo-Achim 
Originally both CWA and CovPass were wrong, but consistent with each other. CWA has been fixed. CovPass has not yet been fixed. So they are inconsistent with each other at the moment. When CovPass is fixed then CWA and CovPass will be consistent with each other again (hopefully!).

**Comment by Ein-Tim** [comment]
> But it seems logical to me that if there are currently no entry restrictions, a valid EU-wide certificate must also suffice for entry! In all other cases, it would mean: 'no business rules' no validity. Or?

No, each country has to provide their rules. Otherwise it will and cannot work. We can't say "this is valid" when in fact it isn't. 
E.g. some European countries require a negative RAT test from less than 24h ago while other require on less than 48h ago.

> However, I see big problems. We are increasingly moving away from 'keep is simple' to 'kindly research on different platforms / in different apps' and make sure that you find / use the right one.
That can not be!

Where exactly do you see this problem? To stay on the topic, do you think it would be better to just show valid if there are no entry rules and thus risking that someone is in the believe that their certificate is valid although it isn't? Yes, this would probably be easier for us, but for the user who is rejected from entering the country, for them it will be just really bad. 

---

> (And yes, it is becoming increasingly impossible to verify whether a topic or parts of a topic have been dealt with somewhere in the vastness of the CWA / CovPass threads or not. At least for people who do not follow this topic 24/7 either.)

I know, even for me it's hard to keep up (-;
And if I or any other community member (like here @MikeMcC399, I hope I can also speak on your behalf here) point to other threads, issues, or resources where this issue was already mentioned, we never mean it in a "bad" way ("Why didn't you search for it", or similar).
We only want to help out and point you to resources that may help you (and others who may be reading by here) to dive even more deeper into the topic so that you by yourself can decide how to proceed. 

Thanks for your involvement & have a nice Saturday evening everyone!

**Comment by Jo-Achim** [comment]
First of all it is clear that none of the apps (CovPass & CWA) should display anything wrong (see below)!

But to get to the point, I am of the opinion that a lack of (business) rules must lead to the validity of the lowest common denominator - namely, the 'complete vaccination' _(1)_. Which means that the valid proof of this must also suffice for entry into the EU country (without defined (business) rules) _(2)_. With which I find the current display in CovPass (1.95.12) to be correct in the result or logical conclusion.

I'll try again in a nutshell:
We have - and this is how I understand the political will (although this is sometimes a bit difficult) - across the EU we have created the lowest common denominator for travel in the 'full vaccination' _(1)_ and certain framework conditions for this. At the same time, it was rightly decided to add further hurdles to this 'lowest common denominator' if necessary; for example, restrictions on having to submit an additional test when entering from high-incidence areas, quarantine or ....

From my understanding, these hurdles / restrictions, if they exist, are to be stored in rules
> **each country has to provide their rules**

, which can then - at least partially - be checked using certificate data. For example, a minimum remaining validity of the checked certificate, the date of the last vaccination, possibly also the existence of a required additional test, etc.

If I am not completely wrong _(1)_ here I come to the above conclusion that if there are no (business) rules or missing components, a valid vaccination certificate ('lowest common denominator') must lead to an entry permit. 
_(1):_ I have noted that under https://www.coronawarn.app/en/faq/#dcc_no_rules as a reason for the changes misinterpretations regarding **incomplete vaccinations** are given.

(For all the things that cannot be checked with the certificate data, the reference to in https://reopen.europa.eu/en integrated in the CWA is correct.)


_(2):_
However, it becomes more complicated if, despite "**each country has to provide their rules**", these actually cannot be used, are empty - because the country in question does not currently want to define any additional restrictions - or parts of the rules are not defined. 
Presumably you focus on this point?
Here, however, the individual countries would be responsible - although this is of little use in case of doubt.

> Where exactly do you see this problem? 

I said that in the paragraph. And it goes without saying that incorrect information must not appear - but I already say above.

-----------

PS: By the way, I think there is no reason to feel "scratched" in any way. We hardly need to discuss the question of who is involved here and how - here I am more of the 'one who rarely looks by'.
However, if abnormalities such as those mentioned above, including a reference to the position that is more correct from the point of view of the beholder, lead to indications “one could have searched and read”, there will soon be hardly any more contributions here apart from the experts. Regardless of such points as 'outdated documentations' and their attention.

------------

In this sense: That should have been it on my part for the time being. 
I'm taking a well-deserved vacation.

**Comment by Ein-Tim** [comment]
@Jo-Achim 

> But to get to the point, I am of the opinion that a lack of (business) rules must lead to the validity of the lowest common denominator - namely, the 'complete vaccination' (1). Which means that the valid proof of this must also suffice for entry into the EU country (without defined (business) rules) (2). With which I find the current display in CovPass (1.95.12) to be correct in the result or logical conclusion.

I disagree here. The problem starts with the definition of "complete vaccination". When is a vaccination complete? In Germany after min. 14 days after the last shot. In Italy after min 14 days after the first shot, but only for 20 days. Every country defines "complete vaccination" different. 

---

> PS: By the way, I think there is no reason to feel "scratched" in any way. We hardly need to discuss the question of who is involved here and how - here I am more of the 'one who rarely looks by'.
However, if abnormalities such as those mentioned above, including a reference to the position that is more correct from the point of view of the beholder, lead to indications “one could have searched and read”, there will soon be hardly any more contributions here apart from the experts. Regardless of such points as 'outdated documentations' and their attention.

No, there is no need to discuss who is involved here, everybody can participate in their way. 
Reg. the "one could have searched and read", I don't want to indicate that. If I point to other threads I know that it's hardly possible for someone not reading by here 24h to keep up with everything. I hope we still get input from normal users too, to not create an "expert bias" where technically everything is "perfect" but the normal user is just lost. 

---

Enjoy your vacation! 🏝

**Comment by MikeMcC399** [comment]
@Jo-Achim
I think that you are basically correct that a full vaccination certificate should allow entry to all EU member states, so I understand how you expect this to be applied by the CWA app. There are however exceptions and that is the problem to apply this to the app results.

---
On https://ec.europa.eu/info/live-work-travel-eu/coronavirus-response/safe-covid-19-vaccines-europeans/eu-digital-covid-certificate_en#how-will-it-help-free-movement it says:

"The EU Digital COVID Certificate will be accepted in all EU Member States. It will help to ensure that restrictions currently in place can be lifted in a coordinated manner.

When travelling, the EU Digital COVID Certificate holder should in principle be exempted from free movement restrictions: Member States should refrain from imposing additional travel restrictions on the holders of an EU Digital COVID Certificate, unless they are necessary and proportionate to safeguard public health. 

In such a case – for instance as a reaction to new variants of concern – that Member State would have to notify the Commission and all other Member States and justify this decision."

I think it is acceptable for the app to err on the side of caution and just say that the certificates can't be checked if there are no electronic business rules available for the destination country. The user is advised in that case to refer to the rules listed in text form on https://reopen.europa.eu and decide manually. For instance for Italy the travel rules are on https://reopen.europa.eu/en/map/ITA/7001 and this site points to another site https://infocovid.viaggiaresicuri.it/index_en.html where travelers can provide additional information for checking.

---
Also, as @Ein-Tim said, countries define complete vaccination status differently. According to https://reopen.europa.eu/en/map/FRA/7001 France says:
Proof of vaccination, which is valid:
- from 7 days after the second dose of the vaccine;
- from 28 days after the single dose of the Janssen/Johnson&Johnson vaccine;
- from 7 days after the first dose of any approved vaccine, for people who recovered from a previous COVID-19 infection.

Germany on the other hand says 14 days from the second dose of a two-series vaccine or 14 days after Johnson&Johnson. There is no wait at all after a vaccination following recovery in Germany.

So the devil lies in the detail unfortunately!

**Comment by Jo-Achim** [comment]
@Ein-Tim, @MikeMcC399,

first of all, thank you for your information.

I do not think that we can discuss the issue here exhaustively and that we will come to a 'general' solution.

However, I would just like to add a few points.
Many of the examples mentioned in which the “devil is in the details” seem to me to be resolvable with the data from the vaccination certificate alone. Because when which vaccination was carried out, how many days have passed since then, etc. can already be found in the digital EU vaccination certificate or calculated or checked from it. And even questions about third vaccination, 'full vaccination protection after first vaccination' etc. could be deduced in my opinion.

This means that different data such as:
>     * from 7 days after the second dose of the vaccine;
>     * from 28 days after the single dose of the Janssen/Johnson&Johnson vaccine;
>     * from 7 days after the first dose of any approved vaccine, for people who recovered from a previous COVID-19 infection.
can be handled by the app - solely from the data / the type of certificate.

Even the question of whether an additional RAT test is necessary or available could be resolved by means of a reference text or even the check whether such a test is known to the CWA.
(It is clear to me that this is quite a challenge in terms of programming. However, I did not formulate the claim to integrate this type of examination into the CWA (especially since it is integrated in CovPass in parallel)).

To illustrate what was said above with a small example: About 14 days ago, both the CWA and CovPass tests agreed that a family member should not have been allowed to enter the Ukraine - the last vaccination was more than 270 days ago.

But of course I have to accept that there may be other / further restrictions that cannot be derived directly or indirectly from the data of the digital EU vaccination certificates. 

For example, if a country, regardless of vaccination status, would require prior travel registration and this would have to be presented at the border, etc. 
In such points, I tend to see the 'devil in the details'.

I already mentioned another point:
> However, it becomes more complicated if, despite "**each country has to provide their rules**", these actually cannot be used, are empty - because the country in question does not currently want to define any additional restrictions - or parts of the rules are not defined.

A possible problem could be added here if states have defined rules, but do not make them available in a timely manner or not at all electronically (and in a standardized manner). These and similar problems could be solved politically at EU level, only.

To cut a long story short: I'm not completely convinced yet. _But I accept that it will be 'a little' more complicated and diverse behind the scenes than is visible from the outside._

At that point we will have to wait a while for further developments. If, despite the corona pandemic, we want to return to the greatest possible and at the same time safe travel freedom, then certain framework conditions will certainly have to change. However, I doubt whether this will make it any easier. Just moving away from a simple number such as the incidence towards a locally (state-specific) combination of incidence and hospital occupancy (keyword: 'hospital traffic light') shows where it is going.

In this sense I remain, Joachim.

PS: We can close this topic if necessary.

**Comment by MikeMcC399** [comment]
@Jo-Achim 
The results on validity checking in Corona-Warn-App and CovPass App for countries with no rules are now aligned:

- Corona-Warn-App 2.10.1 says "Your certificate could not be validated"
- CovPass App 1.165.6 says "Not possible to validate" / "Certificate cannot be validated" (see https://github.com/Digitaler-Impfnachweis/covpass-android/issues/67#issuecomment-934576151)
